app/services/course/assessment/answer/auto_grading_service.rb

```diff
@@ -0,0 +1,37 @@
+class Course::Assessment::Answer::AutoGradingService
+  class << self
+    # Picks the grader for the given grading, then grades into the given
+    # +Course::Assessment::Answer::AutoGrading+ object.
+    #
+    # @param [Course::Assessment::Answer::AutoGrading] auto_grading The object to store grading
```

Comment:
I currently enforce this to be one answer one grading. I'm not sure if there's a need to have one answer graded multiple times.

In any case, this should not be directly called by the client code. I'll abstract this behind the answer model so the answer model handles the job.